table/filter_block.h

```diff
@@ -123,6 +123,29 @@ class FilterBlockReader {
   virtual void CacheDependencies(bool /*pin*/,
                                  const SliceTransform* /*prefix_extractor*/) {}
 
+  virtual bool IsFilterCompatible(
+      const ReadOptions& /*read_options*/, const Slice& /*user_key*/,
+      const Comparator* /*comparator*/) {
+    return true;
+  }
+
+  bool RangeMayExist(const ReadOptions& read_options, const Slice& user_key,
+        const SliceTransform* prefix_extractor, const Comparator* comparator,
+        const Slice& prefix, uint64_t block_offset, const bool no_io,
+        const Slice* const const_ikey_ptr, bool* filter_checked,
+        bool prefix_extractor_changed) {
```

Comment:
`prefix_extractor_changed` is a little bit confusing here to me. Maybe change it to `need_upper_bound_check` or something like that.